Differentiate Oprative and config roles (#607)
ioBroker/ioBroker.docs: docs/en/dev/stateroles.md

The Role information is a very important information and allows Visualization- and Smart-Assistant adapters 
to detect the function of the object and also how/if they relate to other objects in the same channel, device or folder.

## State role types
The following State Role types exist:

### Operative States
Operative states are used to control the normal functionality of a device. A RGB Lamp can have the following three objects (or more) with different roles that belong together:
* `switch` (On/Off)
* `level.color.rgb` with #RRGGBB color code of the lamp
* `level.brightness` with the brightness value

Also the cleaning mode or the room-to-clean of a robotic vacuum cleaner is such an operative state. These states are using the below definition witout any adjustments.

> Please use the most detailed role name psible that prvide the most details (e.g. `level.color.temperature` should be used over `level` for the color tenperature, or `switch.power` is better then `switch` to operate the power of a device).

Reviewer: Passive

Reviewer: "than" not "then"

Author: fixed

> Please use the most detailed role name psible that prvide the most details (e.g. `level.color.temperature` should be used over `level` for the color tenperature, or `switch.power` is better then `switch` to operate the power of a device).

Author: ```suggestion
Please use the most detailed role name possible that provide the most details (e.g. `level.color.temperature` should be used over `level` for the color temperature, or `switch.power` is better than `switch` to operate the power of a device).
```

> Additionally, when using detailed role names (more then one level), it is important not to use the same role twice in a channel of a device.

Author: ```suggestion
Additionally, when using detailed role names (more than one level), it is important not to use the same role twice in a channel of a device.
```

> Additionally, when using detailed role names (more then one level), it is important not to use the same role twice in a channel of a device.

Author: fixed

Different Device templates used for the detecting with the required and optional objects and their roles can be found 
in the [Type-detector repository](https://github.com/ioBroker/ioBroker.type-detector/blob/master/DEVICES.md).

### Configuration/Setting States
States that are configuring further "Non-operative" settings of the devices can also use the below basic role definitions to give more context of the type and usage of the provided value, **but add a ".setting." as second level of the role name**. For example:
* `level.setting.color.temperature` with a 0..100% number can be used to set the "Startup ColorTemperature" of a light bulb
* `switch.setting` (On/Off) coud be used to define setting that can be turned on or off (e.g. child lock functionality)

User Interfaces might use these special roles to determine device settings and show then in a "Settings" dialog for the device, or ignore them.

> Please note: This type of roles were defined in June 2025, so many (older) adapters might not use them. In the future this state type can/should be used when relevant.

Reviewer: "These types of roles"

Author: fixed

> Please note: This type of roles were defined in June 2025, so many (older) adapters might not use them. In the future this state type can/should be used when relevant.

Author: ```suggestion
Please note: These types of roles were defined in June 2025, so many (older) adapters might not use them. In the future this state type can/should be used when relevant.
```
